openvino_xai/explainer/visualizer.py

```diff
@@ -21,9 +21,14 @@
 def resize(saliency_map: np.ndarray, output_size: Tuple[int, int]) -> np.ndarray:
     """Resize saliency map."""
     x = saliency_map.transpose((1, 2, 0))
-    x = cv2.resize(x, output_size[::-1])
-    if x.ndim == 2:
-        return np.expand_dims(x, axis=0)
+
+    resized_channels = []
+    for c in range(x.shape[-1]):
+        # Resize fails for tensors with 700 and more channels (targets=all classes scenario)
```

Comment:
Could you please elaborate about the failure? Some memory issues?
Per-channel resize is very slow, imagine you have 20-30K classes, it might take long. 
One workaround option is to do resize of a tensor with size 500, what do you think?

Reply:
The issue doesn't show the clear memory problem log. It just fails for more than 700+ channels with the log: 
```
cv2.error: OpenCV(4.9.0) /io/opencv/modules/imgproc/src/resize.cpp:3789: error: (-215:Assertion failed) !dsize.empty() in function 'resize'
```
Do you suggest resizing it in batches with 500 channels? Or less 500 channels -> resize at once, more -> resize per-channel?

Reply:
the first one: resizing it in batches with 500 channels